### PR TITLE
enable orientation and empty rows settings in pdf export

### PR DESCRIPTION
**Pull request by triszt4n**
Rewritten part of html file on openingShow, rewritten controller to let the user modify orientation and plus empty rows. (There is a new function for putting empty rows into PDF file)

### CODE REVIEW

File: src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt [hunk 1]
```diff
@@ -408,6 +408,8 @@ class ConfigureController {
     @Throws(FileNotFoundException::class, DocumentException::class)
     fun showOpenings(openingId: Long,
                      type: String?,
+                     page: String?,
+                     emptyrows: Long = 0,
```
Comment (OmTheTurtle) [inline src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt:412]: Does this need to bee of type Long? A basic Int should be enough

File: src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt [hunk 2]
```diff
@@ -408,6 +408,8 @@ class ConfigureController {
     @Throws(FileNotFoundException::class, DocumentException::class)
     fun showOpenings(openingId: Long,
                      type: String?,
+                     page: String?,
```
Comment (OmTheTurtle) [inline src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt:411]: I'd rename this variable to `pageOrientation` to reflect better what it stands for.

File: src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt [hunk 3]
```diff
@@ -435,7 +437,11 @@ class ConfigureController {
         table.setWidths(export.getWidths())
         table.widthPercentage = 100f
         addTableHeader(table, export)
-        addRows(table, export, opening)
+        val added = addRows(table, export, opening)
```
Comment (OmTheTurtle) [inline src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt:440]: Let's rename this to `orderRows`

File: src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt (hunk 1 above)
Comment (triszt4n) [reply src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt]: Just applied all your suggestions in the next commit, thanks!

File: src/main/resources/i18n/messages.properties [hunk 4]
```diff
@@ -281,4 +281,8 @@ lang.export.field-order=Mezők sorrendje
 lang.export.pdf=PDF
 lang.export.csv=CSV
 lang.export.scheme=Séma kiválasztása:
+lang.export.page_orientation=Oldalbeállítás:
+lang.export.landscape=Fekvő
+lang.export.portrait=Álló
+lang.export.empty_rows=Üres sorok hozzáadása:
```
Comment (Gerviba) [inline src/main/resources/i18n/messages.properties:287]: Just to be consistent, please replace the underscore with dashes.

File: src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt [hunk 5]
```diff
@@ -408,15 +408,18 @@ class ConfigureController {
...
+    ): String {
         val opening = openings.getOne(openingId)
         if (cannotEditCircle(opening.circle!!.id!!, request))
             return "redirect:/configure/" + opening.circle!!.id + "?error"
 
         val document = Document()
         val export = ExportType.valueOf(type!!)
 
-        if (export.isPortrait()) {
+        if (pageOrientation.equals("portrait")) {
```
Comment (Gerviba) [inline src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt:422]: Don't use magic constants. They can cause you headache if you want to change or implement a new related feature and you need to look for all the possible values. Use `enum classes` if the set of possible values are predetermined and `const val` if they're no not. You can also avoid mistyping by that way.

File: src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt [hunk 6]
```diff
@@ -472,6 +479,18 @@ class ConfigureController {
...
             }
         }
+        return ordersList.size;
+    }
+
+    private fun addEmptyRows(table: PdfPTable, export: ExportType, emptyRows: Int) {
+        for (i in 1..emptyRows) {
+            export.fields.forEach { _ ->
+                val cell = PdfPCell()
+                cell.phrase = Phrase(" ") // <-- Needs a whitespace
```
Comment (Gerviba) [inline src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt:489]: Why does it need a white space? I mean I have some tips, but if you make a comment make it clear for anyone.
For example:
```
// NOTE: Empty phrases are ignored. Therefore at least a whitespace character is needed to be added.
```

File: src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt [hunk 7]
```diff
@@ -431,11 +434,15 @@ class ConfigureController {
...
-        val table = PdfPTable(export.getHeader().size)
-        table.setWidths(export.getWidths())
+        val table = PdfPTable(export.header.size)
+        table.setWidths(export.widths)
         table.widthPercentage = 100f
         addTableHeader(table, export)
-        addRows(table, export, opening)
+        val orderRows = addOrderRows(table, export, opening)
+        addEmptyRows(table, export, emptyRows)
+        if ((emptyRows + orderRows) == 0) { // <-- In case of empty document
```
Comment (Gerviba) [inline src/main/kotlin/hu/kirdev/schpincer/web/ConfigureController.kt:443]: If you want to clarify a condition you can always extract it to a method. 
Tip: Select the condition, `Ctrl+Alt+M`, enter something like `isDocumentEmpty`, hit enter

File: src/main/resources/i18n/messages.properties (hunk 4 above)
Comment (triszt4n) [reply src/main/resources/i18n/messages.properties]: Applied all your suggestions, thank you for the tips and tricks, all very useful!

File: src/main/resources/templates/openingShow.html [hunk 8]
```diff
@@ -240,9 +259,9 @@ <h3 th:text="#{lang.export.title-table}">Table export</h3>
 						<td th:text="#{lang.export.order-by}">Order by</td>
 						<td>
 							<select name="orderby">
-								<option th:text="#{lang.export.order.absolute}" th:value="${T(hu.gerviba.webschop.service.OrderService).ORDER_ABSOLUTE}">Absolute order</option>
-								<option th:text="#{lang.export.order.grouped}" th:value="${T(hu.gerviba.webschop.service.OrderService).ORDER_GROUPED}">Grouped timelines</option>
-								<option th:text="#{lang.export.order.rooms}" th:value="${T(hu.gerviba.webschop.service.OrderService).ORDER_ROOMS}">Grouped timelines and rooms</option>
+								<option th:text="#{lang.export.order.absolute}" th:value="${T(hu.kirdev.schpincer.service.OrderServiceKt).ORDER_ABSOLUTE}">Absolute order</option>
+								<option th:text="#{lang.export.order.grouped}" th:value="${T(hu.kirdev.schpincer.service.OrderServiceKt).ORDER_GROUPED}">Grouped timelines</option>
+								<option th:text="#{lang.export.order.rooms}" th:value="${T(hu.kirdev.schpincer.service.OrderServiceKt).ORDER_ROOMS}">Grouped timelines and rooms</option>
```
Comment (triszt4n) [inline src/main/resources/templates/openingShow.html:264]:
```suggestion
								<option th:text="#{lang.export.order.absolute}" th:value="${T(hu.kirdev.schpincer.service.OrderStrategy).ORDER_ABSOLUTE.representation}">Absolute order</option>
								<option th:text="#{lang.export.order.grouped}" th:value="${T(hu.kirdev.schpincer.service.OrderStrategy).ORDER_GROUPED.representation}">Grouped timelines</option>
								<option th:text="#{lang.export.order.rooms}" th:value="${T(hu.kirdev.schpincer.service.OrderStrategy).ORDER_ROOMS.representation}">Grouped timelines and rooms</option>
```

**Pull request merged**